addons/link_tracker/controller/main.py

```diff
@@ -3,17 +3,21 @@
 
 from odoo import http
 from odoo.http import request
+from odoo.tools.misc import is_a_bot
 
 
 class LinkTracker(http.Controller):
 
     @http.route('/r/<string:code>', type='http', auth='public', website=True)
     def full_url_redirect(self, code, **post):
-        country_code = request.session.geoip and request.session.geoip.get('country_code') or False
-        request.env['link.tracker.click'].sudo().add_click(
-            code,
-            ip=request.httprequest.remote_addr,
-            country_code=country_code
-        )
+        if not is_a_bot(request):
+            country_code = request.session.geoip and request.session.geoip.get(
+                'country_code') or False
+            request.env['link.tracker.click'].sudo().add_click(
+                code,
+                ip=request.httprequest.remote_addr,
+                user_agent=request.httprequest.user_agent,
+                country_code=country_code,
+            )
         redirect_url = request.env['link.tracker'].get_url_from_code(code)
-        return request.redirect(redirect_url or '', code=301, local=False)
+        return request.redirect(redirect_url or '/404.html', code=301, local=False)
```

Comment:
where is `/404.html` defined ? (do the same as "/qsdqsdqsd" so it's better to raise a NotFound error I think)